test/e2e/network/ingress.go

```diff
@@ -733,6 +733,148 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 			By("Scale down number of backends to 2")
 			scaleAndValidateExposedNEG(3)
 		})
+
+		It("should not delete NEGs when switching between Ingress-referenced NEG and standalone NEG", func() {
+			By("Create a basic HTTP ingress using standalone NEG")
+			jig.CreateIngress(filepath.Join(ingress.IngressManifestPath, "neg-exposed"), ns, map[string]string{}, map[string]string{})
+			jig.WaitForIngress(true)
+			usingNEG, err := gceController.BackendServiceUsingNEG(jig.GetServicePorts(false))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(usingNEG).To(BeTrue())
+
+			name := "hostname"
+			svc, err := f.ClientSet.CoreV1().Services(ns).Get(name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			var status ingress.NegStatus
+			v, ok := svc.Annotations[ingress.NEGStatusAnnotation]
+			Expect(ok).To(BeTrue())
+
+			err = json.Unmarshal([]byte(v), &status)
+			Expect(err).NotTo(HaveOccurred())
+			framework.Logf("Got %v: %v", ingress.NEGStatusAnnotation, v)
+
+			// Expect 2 NEGs to be created based on the test setup (neg-exposed)
+			if len(status.NetworkEndpointGroups) != 2 {
+				framework.Logf("Expected 2 NEGs, got %d", len(status.NetworkEndpointGroups))
+				Expect(len(status.NetworkEndpointGroups)).To(Equal(2))
+			}
+
+			// Disable NEG exposed but enable NEG with Ingress. Check NEGs still exist
+			By("Switch backend service to use Ingress-exposed NEG")
+			svcList, err := f.ClientSet.CoreV1().Services(ns).List(metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			for _, svc := range svcList.Items {
+				svc.Annotations[ingress.NEGAnnotation] = `{"ingress": true}`
+				_, err = f.ClientSet.CoreV1().Services(ns).Update(&svc)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			wait.Poll(5*time.Second, framework.LoadBalancerPollTimeout, func() (bool, error) {
```

Comment:
Probably worth packaging this into a function to verify every step of the way. 

The verification may be a bit different:
1. Poll for some time and check if something happens and then stop polling
2. Poll for some time and check if something does not happen, if it happens then failed (This is to make sure NEG was not deleted)